File: Javascript/hollow triangle.js
function hollowTriangle(n){
  let num = n - 2
  let first = [`${"_".repeat(n - 1)}#${"_".repeat(n - 1)}`]
  let middle = Array.from({length: num}, (_, i) => `${"_".repeat(num - i)}#${"_".repeat(i * 2 + 1)}#${"_".repeat(num - i)}`)
  let last = ["#".repeat(n * 2 - 1)]
  return n == 1 ? first : first.concat(middle).concat(last)
}

const { assert } = require("chai")

describe("Tests", () => {
  it("test", () => {
    assert.deepEqual(hollowTriangle(2),['_#_', '###']);
    assert.deepEqual(hollowTriangle(4),['___#___', '__#_#__', '_#___#_', '#######']);
    assert.deepEqual(hollowTriangle(6),['_____#_____', '____#_#____', '___#___#___', '__#_____#__', '_#_______#_', '###########']);
    assert.deepEqual(hollowTriangle(1),['#']);
    assert.deepEqual(hollowTriangle(3),['__#__', '_#_#_', '#####']);
  });
});

⬇️⬇️

/* Example(6)：
['_____#_____', 
 '____#_#____', 
 '___#___#___', 
 '__#_____#__', 
 '_#_______#_', 
 '###########']
*/

File: python/hollow triangle.py
# https://www.codewars.com/kata/57819b700a8eb2d6b00002ab

def hollow_triangle(n):
    top = [f'{"_" * (n - 1)}#{"_" * (n - 1)}']
    middle = [f'{"_" * (n - 2 - i)}#{"_" * (i * 2 + 1)}#{"_" * (n - 2 - i)}' for i in range(n - 2)]
    bottom = [f'{"#" * (n * 2 - 1)}']
    return top + middle + bottom
 
test.assert_equals(hollow_triangle(2),['_#_', '###'])
test.assert_equals(hollow_triangle(4),['___#___', '__#_#__', '_#___#_', '#######'])
test.assert_equals(hollow_triangle(6),['_____#_____', '____#_#____', '___#___#___', '__#_____#__', '_#_______#_', '###########'])
test.assert_equals(hollow_triangle(1),['#'])
test.assert_equals(hollow_triangle(3),['__#__', '_#_#_', '#####'])
test.assert_equals(hollow_triangle(9),['________#________', '_______#_#_______', '______#___#______', '_____#_____#_____', '____#_______#____', '___#_________#___', '__#___________#__', '_#_____________#_', '#################'])

⬇️⬇️

"""Example (n = 6)：
['_____#_____', 
 '____#_#____', 
 '___#___#___', 
 '__#_____#__', 
 '_#_______#_', 
 '###########']
"""
